apps/docs.blocksense.network/components/common/Dialog.tsx

```diff
@@ -0,0 +1,243 @@
+import React, {
+  useState,
+  useRef,
+  useEffect,
+  createContext,
+  useContext,
+  HTMLAttributes,
+  ReactNode,
+  RefObject,
+  MouseEvent,
+} from 'react';
+
+import { cn } from '@/lib/utils';
+import { Icon } from '@blocksense/ui/Icon';
+import { Button } from '@blocksense/ui/Button';
+
+interface DialogContextValue {
+  isOpen: boolean;
+  setOpen: (open: boolean) => void;
+  toggle: () => void;
+  dialogRef: RefObject<HTMLDivElement>;
+}
+const DialogContext = createContext<DialogContextValue>(
+  {} as DialogContextValue,
+);
+
+export const Dialog = ({
+  children,
+  isOpen,
+  onClose,
+}: {
+  children: ReactNode;
+  isOpen: boolean;
+  onClose?: () => void;
+}) => {
+  const dialogRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    const handleClickOutside = (event: Event) => {
+      if (
+        dialogRef.current &&
+        !dialogRef.current.contains(event.target as Node)
+      ) {
+        if (onClose) onClose();
+      }
+    };
+    const handleKeyDown = (event: KeyboardEvent) => {
+      if (event.key === 'Escape') {
+        if (onClose) onClose();
+      }
+    };
+    if (isOpen) {
+      document.body.classList.add('overflow-hidden');
+      document.addEventListener('mousedown', handleClickOutside);
+      document.addEventListener('keydown', handleKeyDown);
+    } else {
+      document.body.classList.remove('overflow-hidden');
+    }
+    return () => {
+      document.body.classList.remove('overflow-hidden');
+      document.removeEventListener('mousedown', handleClickOutside);
+      document.removeEventListener('keydown', handleKeyDown);
+    };
+  }, [isOpen, onClose]);
```

Comment:
```suggestion
useEffect(() => {
  if (!isOpen) {
    document.body.classList.remove('overflow-hidden');
    return;
  }

  document.body.classList.add('overflow-hidden');

  const handleClickOutside = (event: MouseEvent) => {
    if (dialogRef.current && !dialogRef.current.contains(event.target as Node)) {
      onClose?.();
    }
  };

  const handleKeyDown = (event: KeyboardEvent) => {
    if (event.key === 'Escape') {
      onClose?.();
    }
  };

  document.addEventListener('mousedown', handleClickOutside);
  document.addEventListener('keydown', handleKeyDown);

  return () => {
    document.body.classList.remove('overflow-hidden');
    document.removeEventListener('mousedown', handleClickOutside);
    document.removeEventListener('keydown', handleKeyDown);
  };
}, [isOpen, onClose, dialogRef]);
```
- If the dialog isn’t open, we immediately remove overflow-hidden and return. This keeps the code simpler and avoids unnecessary event bindings
- We attach event listeners only when the dialog is open and remove them (along with the hidden overflow) in the cleanup function. This ensures consistent cleanup every time the dialog closes.